src/cp/param.h

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: 2025 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2025 Siddharth Mishra <admin@brightprogrammer.in>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#ifndef CPDEM_param_H
+#define CPDEM_param_H
+
+#include "cp/vec.h"
+#include "demangler_util.h"
+
+/**
+ * Represents a template or function parameter.
+ */
+typedef struct {
+    DemString *name;
+    DemString *suffix;
+    DemString *prefix;
+} Param;
+
+/**
+ * Initialize clone from p_src to p_dst.
+ *
+ * \param p_dst : Pointer to Param object to init clone into.
+ * \param p_src : Pointer to Param object to init clone of.
+ *
+ * \return p_dst on success.
+ * \return NULL otherwise.
+ * */
+#define param_init_clone(p_dst, p_src)                                                             \
+    ((param_init (p_dst) && (p_src)) ? (dem_string_concat ((p_dst)->name, (p_src)->name),          \
+                                        dem_string_concat ((p_dst)->suffix, (p_src)->suffix),      \
+                                        dem_string_concat ((p_dst)->prefix, (p_src)->prefix),      \
+                                        (p_dst)) :                                                 \
+                                       NULL)
+
+/**
+ * Init object make it usable with other functions (macros).
+ *
+ * \param p : Pointer to function parameter.
+ *
+ * \return p on success.
+ * \return NULL otherwise.
+ * */
+#define param_init(p)                                                                              \
+    ((p) ? (((p)->name = dem_string_new()),                                                        \
+            ((p)->suffix = dem_string_new()),                                                      \
+            ((p)->prefix = dem_string_new()),                                                      \
+            (p)) :                                                                                 \
+           NULL)
```

Comment:
same here

Reply:
We can convert these after the rewrite is complete, it won't hurt to change them and the code will be going into static mode.